ui/app/routes/jobs/job/definition.js

```diff
@@ -31,7 +31,12 @@ export default class DefinitionRoute extends Route {
 
   setupController(controller, model) {
     super.setupController(controller, model);
-    controller.view =
-      model?.specification && !controller.view ? 'job-spec' : 'full-definition';
+
+    const view = controller.view
+      ? controller.view
+      : model?.specification
+      ? 'job-spec'
+      : 'full-definition';
+    controller.view = view;
```

Comment:
This makes me want to further handle these as `@tracked` values with defaults and possibly a constructor in the definition controller. I'd be worried as a future editor that I'd miss where this logic lives.

Reply:
The query parameter is a `tracked` property! However, for instantiating the value on constructing the controller, this logic must be handled in the `setupController` hook because the `model` isn't available.

Reply:
There are a few things using the model in the controller currently — we're doing

```
@alias('model.job') job;
@alias('model.definition') definition;
```
there for example. 

Is there something that prevents `specification` from being referenced?

Reply:
This brings up a discussion that I wish Ember had a "cookbook" for. I had these same thoughts running the process and tried implementing almost exactly what you're talking about. Specifically I wanted a recipe for "how do I initialize a query parameter?".

If we're initializing query parameters: we need a resolved model.

The `setupController` hook is called after the model hook resolves, so you have access to both the resolved model and any query parameters at this stage.

We also need to support one more thing:

Responding to query parameter changes: If you need to perform some action whenever a query parameter changes, you can use the `setupController` hook in combination with queryParams configuration to respond to those changes.

Another area when we'd want to use this would be setting up properties on the controller based on query parameters: when you want to set up properties on the controller that depend on the current value of the query parameters (as I normally refer to as "derived state".